cordova-lib/spec-cordova/ConfigParser.spec.js

```diff
@@ -114,78 +114,94 @@ describe('config.xml parser', function () {
-                var feature = cfg.getFeature('org.apache.cordova.featurewithurl');
-                expect(feature.url).toEqual('http://cordova.apache.org/featurewithurl');
-                expect(feature.params).toBeDefined();
-                expect(feature.params.url).toBeDefined();
-                expect(feature.params.url).toEqual('http://cordova.apache.org/featurewithurl');
-            });
-            it('should read feature with version and set \'version\' param', function(){
-                var feature = cfg.getFeature('org.apache.cordova.featurewithversion');
-                expect(feature.version).toEqual('1.1.1');
-                expect(feature.params).toBeDefined();
-                expect(feature.params.version).toBeDefined();
-                expect(feature.params.version).toEqual('1.1.1');
-            });
-            it('should read feature variables', function () {
-                var feature = cfg.getFeature('org.apache.cordova.featurewithvars');
-                expect(feature.variables).toBeDefined();
-                expect(feature.variables.var).toBeDefined();
-                expect(feature.variables.var).toEqual('varvalue');
-            });
-            it('should allow adding a new feature', function(){
-                cfg.addFeature('myfeature');
-                var features = cfg.doc.findall('feature');
-                var featureNames = features.map(function(feature){
-                    return feature.attrib.name;
+            it('should read plugin given id', function(){
+                var plugin = cfg.getPlugin('org.apache.cordova.justafeature');
+                expect(plugin).toBeDefined();
+                expect(plugin.name).toEqual('org.apache.cordova.justafeature');
+                expect(plugin.variables).toBeDefined();
+            });
+            it('should not read plugin given undefined id', function(){
+                var plugin = cfg.getPlugin('org.apache.cordova.undefinedfeature');
+                expect(plugin).not.toBeDefined();
+            });
+            it('should read plugin with src', function(){
+                var plugin = cfg.getPlugin('org.apache.cordova.featurewithurl');
+                expect(plugin.src).toEqual('http://cordova.apache.org/featurewithurl');
+            });
+            it('should read plugin with version', function(){
+                var plugin = cfg.getPlugin('org.apache.cordova.featurewithversion');
+                expect(plugin.version).toEqual('1.1.1');
+            });
+            it('should read plugin variables', function () {
+                var plugin = cfg.getPlugin('org.apache.cordova.featurewithvars');
+                expect(plugin.variables).toBeDefined();
+                expect(plugin.variables.var).toBeDefined();
+                expect(plugin.variables.var).toEqual('varvalue');
+            });
+            it('should allow adding a new plugin', function(){
+                cfg.addPlugin({name:'myfeature'});
```

Comment:
This test does not seem to test all the way to the filesystem. 

You should use cfg.write() here to flush things to the filesystem, then reload from the filesystem using the constructor.
As it currently stands, we are just changing things in memory. But I suppose the goal is to test the end-to-end scenario: from memory to filesystem and back.